### ISSUE TITLE
rethinkdb-dump exits without finishing

### ISSUE THREAD

**Issue by bsharpe** [rethinkdb/rethinkdb#6056]
```
Zipping export directory...
Traceback (most recent call last):
  File "/usr/local/bin/rethinkdb-dump", line 9, in <module>
    load_entry_point('rethinkdb==2.3.0.post5', 'console_scripts', 'rethinkdb-dump')()
  File "/usr/local/lib/python2.7/dist-packages/rethinkdb/_dump.py", line 203, in main
    run_rethinkdb_export(options)
  File "/usr/local/lib/python2.7/dist-packages/rethinkdb/_dump.py", line 184, in run_rethinkdb_export
    do_zip(temp_dir, options)
  File "/usr/local/lib/python2.7/dist-packages/rethinkdb/_dump.py", line 166, in do_zip
    archive.close()
AttributeError: 'NoneType' object has no attribute 'close'
```

```
> rethinkdb --version
rethinkdb 2.3.4~0trusty (GCC 4.8.2)
> uname -a
Linux redis-1 4.2.0-38-generic #45~14.04.1-Ubuntu SMP Thu Jun 9 09:27:51 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by bsharpe** [comment]
nevermind... it was a permissions thing...

**Comment by ducktype** [comment]
The error still wrong... Should say explicitly that there was a permission error while writing the output file!

**Comment by danielmewes** [comment]
Re-opening because the error message is bad.

**Comment by larkost** [comment]
It looks like the actual error message was hidden behind an over-eager `finally` block. This is fixed in `next` with 52468baa.
